program-runtime/src/compute_budget.rs

```diff
@@ -37,6 +37,8 @@ pub struct ComputeBudget {
     pub invoke_units: u64,
     /// Maximum cross-program invocation depth allowed
     pub max_invoke_depth: usize,
+    /// Maximum cross-program invocation and instructions per transaction
+    pub max_instruction_trace_length: usize,
```

Comment:
Maybe at this level it would be clearer to call this `max_instructions`? "instruction_trace" is kinda an implementation detail (and until recently, I did not know what it was :P)

```
/// Maximum number of instruction allowed. Includes top-level and CPI invocations.
pub max_instructions: usize
```

Reply:
Originally, I wanted to call it that, but then it occurred to me that "instruction" can mean two things:
- Input instructions as they are defined in the message and which form the transaction-level instructions
- Output instructions which include the above plus CPI instructions which were generated at runtime

So instruction trace makes it clear that it refers to the output generated by the program runtime.

Reply:
Fair enough, convinced!